phaseshifts/atorb.py

```
@@ -858,3 +1255,358 @@ def calculate_Q_density(**kwargs):
+            with open(atorb_file, "w") as f:
+                f.write(io.getvalue())
+            successful = True
+        except Exception as e:
+            raise e
+        finally:
+            # clean up
+            io.close()
+        return atorb_file if successful else None
+
+    @staticmethod
+    def calculate_Q_density(
```

Ruff (0.14.10)

Missing return type annotation for staticmethod `calculate_Q_density`

(ANN205)